src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -222,39 +229,31 @@ off_t ReadBufferFromRemoteFSGather::seek(off_t offset, int whence)
     if (whence != SEEK_SET)
         throw Exception(ErrorCodes::CANNOT_SEEK_THROUGH_FILE, "Only SEEK_SET mode is allowed.");
 
-    if (use_external_buffer)
+    if (!working_buffer.empty()
```

Comment:
> [...] do ignore in this seek, which will not break because you have ensureInternalBuffer, e.g. a new buffer will be allocated

No, the `ignore()` is inside an `if (... && !internal_buffer.empty())`